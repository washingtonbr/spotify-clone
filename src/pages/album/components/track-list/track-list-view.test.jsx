import React from 'react';
import renderer from 'react-test-renderer';
import TrackListView from './track-list-view';

jest.mock('../../../../utils/time', () => ({
  millisToMinutesAndSeconds: jest.fn(),
}));

const mockTracks = [
  {
    duration_ms: 173893,
    id: '7ccTcabbJlCJiIqtrSSwBk',
    name: 'Premonition - Intro',
  },
  {
    duration_ms: 216600,
    id: '0q2vG0UVuy6ajjcpkQHdZM',
    name: 'Unaccommodating (feat. Young M.A)',
  },
  {
    duration_ms: 234653,
    id: '0GQ5bFTVFFKpwNPc7KwQnB',
    name: 'You Gonâ€™ Learn (feat. Royce Da 5\'9" & White Gold)',
  },
  {
    duration_ms: 30133,
    id: '4EmunTy7kNBYQivOa8F6b8',
    name: 'Alfred - Interlude',
  },
  {
    duration_ms: 177693,
    id: '0VSzREd1OjEWJ9tXoFHRQH',
    name: 'Those Kinda Nights (feat. Ed Sheeran)',
  },
  {
    duration_ms: 194706,
    id: '7Dj2wqPYXfSRFwa6qyOztr',
    name: 'In Too Deep',
  },
  {
    duration_ms: 210800,
    id: '7FIWs0pqAYbP91WWM0vlTQ',
    name: 'Godzilla (feat. Juice WRLD)',
  },
  {
    duration_ms: 337146,
    id: '5SiZJoLXp3WOl3J4C8IK0d',
    name: 'Darkness',
  },
  {
    duration_ms: 265946,
    id: '5CpsIb7SMFHjmAG8RvIWfq',
    name: 'Leaving Heaven (feat. Skylar Grey)',
  },
  {
    duration_ms: 286960,
    id: '5mZXWEH2eh8zMZGCxT5aW0',
    name: 'Yah Yah (feat. Royce Da 5\'9", Black Thought, Q-Tip & Denaun)',
  },
];

describe('TrackListView', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <TrackListView
          tracks={mockTracks}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
