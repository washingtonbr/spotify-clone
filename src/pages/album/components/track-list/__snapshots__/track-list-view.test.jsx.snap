// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackListView renders correctly 1`] = `
<div
  className="sc-AxjAm StDqN"
>
  <ul
    className="sc-AxirZ efLPWf"
  >
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        1
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Premonition - Intro
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        2
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Unaccommodating (feat. Young M.A)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        3
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        You Gonâ€™ Learn (feat. Royce Da 5'9" & White Gold)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        4
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Alfred - Interlude
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        5
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Those Kinda Nights (feat. Ed Sheeran)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        6
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        In Too Deep
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        7
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Godzilla (feat. Juice WRLD)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        8
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Darkness
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        9
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Leaving Heaven (feat. Skylar Grey)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
    <li
      className="sc-AxiKw bTjmim"
    >
      <div
        className="sc-AxhCb etchzD"
      >
        10
      </div>
      <div
        className="sc-AxhUy krXQjZ"
      >
        Yah Yah (feat. Royce Da 5'9", Black Thought, Q-Tip & Denaun)
      </div>
      <div
        className="sc-AxgMl cXMoIt"
      />
    </li>
  </ul>
</div>
`;
